backup/src/main/java/io/camunda/zeebe/backup/processing/CheckpointRecordsProcessor.java

```diff
@@ -80,4 +95,13 @@ public ProcessingResult onProcessingError(
     // from making progress.
     throw new RuntimeException(processingException);
   }
+
+  /**
+   * Registers a listener.
+   *
+   * @param checkpointListener
+   */
+  public void addCheckpointListener(final CheckpointListener checkpointListener) {
```

Comment:
:x: If I remember correctly (and this matches with the original issue description), we need to call the listener here directly with the current checkpoint id.

Reply:
:+1: I had added it and then removed it :) But, you are right. It is safer to notify when registering to ensure that it doesn't miss an update. I will update it.